bigquery/google/cloud/bigquery/client.py

```diff
@@ -1929,6 +1935,60 @@ def list_rows(
         )
         return row_iterator
 
+    def schema_from_json(self, file_or_path):
+        """Takes a file object or file path that contains json that describes
+            a table schema.
+
+            Returns:
+                List of schema field objects.
+        """
+        file_obj = None
+        json_data = None
+        schema_field_list = list()
+
+        if isinstance(file_or_path, io.IOBase):
+            file_obj = file_or_path
+        else:
+            try:
+                file_obj = open(file_or_path)
```

Comment:
An important difference with this path is that we *must close* `file_obj` if we open it based on a path, but we *must **not** close* `file_obj` if we are given a file-like object.

I recommend using a `with` statement when we are provided a path to make sure we always close it. You may want to refactor this to add a `_schema_from_json_file_object()` private helper method that does the loading part without the file closing part.

Reply:
Added helper method per recommendation along with the statement to close the file.